org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/WorkspaceSymbolHandler.java

```diff
@@ -229,4 +137,129 @@ public SearchSymbolParams(String query, String projectName) {
 			this.projectName = projectName;
 		}
 	}
-}
\ No newline at end of file
+
+	private static class WorkspaceSymbolTypeRequestor extends TypeNameMatchRequestor {
+		private ArrayList<SymbolInformation> symbols;
+		private int maxResults;
+		private boolean sourceOnly;
+		private boolean isSymbolTagSupported;
+		private IProgressMonitor monitor;
+
+		public WorkspaceSymbolTypeRequestor(ArrayList<SymbolInformation> symbols, int maxResults, boolean sourceOnly, boolean isSymbolTagSupported, IProgressMonitor monitor) {
+			this.symbols = symbols;
+			this.maxResults = maxResults;
+			this.sourceOnly = sourceOnly;
+			this.isSymbolTagSupported = isSymbolTagSupported;
+			this.monitor = monitor;
+		}
+
+		@Override
+		public void acceptTypeNameMatch(TypeNameMatch match) {
+			try {
+				if (maxResults > 0 && symbols.size() >= maxResults) {
+					return;

```

Comment:
Should we set the monitor to cancel state as well? Like line 189.

Reply:
Yup, that makes sense. Note that the cancellation isn't sent back to the client (resulting in no results) because we catch it in the `OperationCanceledException` try-catch. This just prevents us from having to continue evaluating matches when we've reached the limit.

Note that outside of testing, `maxResults` is always `0`, so it's never triggered.